tests/test_temporal_avg.py

```diff
@@ -0,0 +1,3207 @@
+        ds = self.ds.copy()
+
+        # Make the time coordinates uncentered.
+        uncentered_time = np.array(
+            [
+                "2000-01-31T12:00:00.000000000",
+                "2000-02-29T12:00:00.000000000",
+                "2000-03-31T12:00:00.000000000",
+                "2000-04-30T00:00:00.000000000",
+                "2000-05-31T12:00:00.000000000",
+                "2000-06-30T00:00:00.000000000",
+                "2000-07-31T12:00:00.000000000",
+                "2000-08-31T12:00:00.000000000",
+                "2000-09-30T00:00:00.000000000",
+                "2000-10-16T12:00:00.000000000",
+                "2000-11-30T00:00:00.000000000",
+                "2000-12-31T12:00:00.000000000",
+                "2001-01-31T12:00:00.000000000",
+                "2001-02-28T00:00:00.000000000",
+                "2001-12-31T12:00:00.000000000",
+            ],
+            dtype="datetime64[ns]",
+        )
+        ds.time.data[:] = uncentered_time
+
+        # Set object attrs required to test the method.
+        ds.temporal._time_bounds = ds.time_bnds.copy()
+
+        # Compare result of the method against the expected.
+        expected = ds.copy()
+        expected_time_data = np.array(
+            [
+                "2000-01-16T12:00:00.000000000",
+                "2000-02-15T12:00:00.000000000",
+                "2000-03-16T12:00:00.000000000",
+                "2000-04-16T00:00:00.000000000",
+                "2000-05-16T12:00:00.000000000",
+                "2000-06-16T00:00:00.000000000",
+                "2000-07-16T12:00:00.000000000",
+                "2000-08-16T12:00:00.000000000",
+                "2000-09-16T00:00:00.000000000",
+                "2000-10-16T12:00:00.000000000",
+                "2000-11-16T00:00:00.000000000",
+                "2000-12-16T12:00:00.000000000",
+                "2001-01-16T12:00:00.000000000",
+                "2001-02-15T00:00:00.000000000",
+                "2001-12-16T12:00:00.000000000",
+            ],
+            dtype="datetime64[ns]",
+        )
+        expected = expected.assign_coords(
+            {
+                "time": xr.DataArray(
+                    name="time",
+                    data=expected_time_data,
+                    coords={"time": expected_time_data},
+                    dims="time",
+                    attrs={
+                        "long_name": "time",
+                        "standard_name": "time",
+                        "axis": "T",
+                        "bounds": "time_bnds",
+                    },
+                )
+            }
+        )
+        # Update time bounds with centered time coordinates.
+        time_bounds = ds.time_bnds.copy()
+        time_bounds["time"] = expected.time
+        expected["time_bnds"] = time_bounds
+
+        # Compare result of the method against the expected.
+        result = ds.temporal.center_times(ds)
+        assert result.identical(expected)
+
+
```

Comment:
```suggestion


# TODO: Update TestAverager tests to use other numbers rather than 1's for
# better test reliability and accuracy. This may require subsetting.
```